packages/atomic/src/components/facets/facet/base-facet.tsx

```diff
@@ -1,82 +1,81 @@
-import {Component, h, Event, EventEmitter, State, Prop} from '@stencil/core';
+import {h} from '@stencil/core';
 
 import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
 
-@Component({
-  tag: 'base-facet',
-  styleUrl: 'base-facet.pcss',
-  shadow: false,
-})
-export class BaseFacet {
-  @State() public isExpanded: Boolean = false;
-  @Event() public deselectAll!: EventEmitter<void>;
-  @Prop() public label!: string;
-  @Prop() public hasActiveValues!: boolean;
+export interface BaseFacetState {
```

Comment:
Each base component implements BaseFacetState and then passes itself into the controller, the controller is then passed into the functional component which can now mutate state on its parent.